docs/hardwareengagement/index.md

```diff
@@ -0,0 +1,34 @@
+# Hardware Engagement
+
+In this document, we will share our steps as how to engagement kepler with a specific hardware device. Considering there are many different hardware devices, we will just write down major steps here as current stage.
+You are able to take this out as a todo list and step by step to make kepler engage with your own hardware device.
+
+## Stage 0 Proof
+In this Stage, we will focus on basic data can be collected by golang, and you can build your own kepler which running on your device well. The following steps can running in parallel.
+### Binary build and container build.
+Currently kepler container image is from a GPU image to support GPU case. Considering a general case for IOT device. You may need to build kepler from UBI image. We recommand you following steps below to setup a local build env and try to build.
+
+1. Find a linux OS.
+1. Install kepler dependencies as ebpf golang(BCC), linux header and build kepler(from main branch or latest release branch) binary.
+1. (Optional)Modify [dockerfile](https://github.com/sustainable-computing-io/kepler/tree/main/build) to build the contaienr image.
+
+### Power consumption API.
+Currently, we use power consumption API as RAPL or ACPI. For some of the devices, you may need to find your own way to get power consumption, and implement in golang for kepler usage. For further plan, please ref [here](https://github.com/sustainable-computing-io/kepler/issues/644)
```

Comment:
After incorporating the redfish (PR #734), we will need to revise this information.

Additionally, we should mention in this section that we classify the components we gather power from, and each component may have a distinct power source. The extension should be implemented either within the class to accommodate a new component or within the power sources section, depending on the nature of the change.